### PR TITLE
TOOLS-2859 Restore timeseries collections

### PR DESCRIPTION
**Pull request by tfogo**
Restores timeseries collections as dumped by TOOLS-2857. Also correctly includes/excludes timeseries collections. Does not include renaming which is covered by TOOLS-2860.

Based off the TOOLS-2857-dump branch for ease of review.

### CODE REVIEW

File: mongorestore/mongorestore_test.go [hunk 1]
```diff
@@ -1585,3 +1586,338 @@ func TestCreateIndexes(t *testing.T) {
...
+			indexes, err := testdb.Collection("foo_ts").Indexes().List(ctx)
+			defer indexes.Close(ctx)
+			So(err, ShouldBeNil)
+
+			numIndexes := 0
+			for indexes.Next(ctx) {
+				numIndexes++
+			}
+
+			So(numIndexes, ShouldEqual, 0)
```
Comment (huan-Mongo) [inline mongorestore/mongorestore_test.go:1920]: Maybe also check clustered index with `listCollections` cmd?

**Review by huan-Mongo — APPROVED** [review]
LGTM! Just one minor suggestion on the test.

**Comment by huan-Mongo** [comment]
btw @tfogo , I noticed the evergreen patch run was not trigged by the PR and it seems the webhook request was failed last time from the github setting. Do you know is there anything we can do to make the webhook work?

**Comment by tfogo** [comment]
@huan-Mongo I made the changes you suggested. I think the evg patch didn't run because the base branch isn't master. Here's a patch I triggered: https://spruce.mongodb.com/version/60d088cbc9ec4470b6bda1bb/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

Still a lot of failing js tests but I don't think that's related to this PR.

**Review by rychipman — APPROVED** [review]
one small nit which you can feel free to change or ignore -- LGTM either way!

**Review by huan-Mongo — APPROVED** [review]
Cool! LGTM!

**Pull request merged**